plugins/jobs/server/models/job.py

```diff
@@ -513,3 +534,30 @@ def getAllTypesAndStatuses(self, user):
         types = self.collection.distinct('type', query)
         statuses = self.collection.distinct('status', query)
         return {'types': types, 'statuses': statuses}
+
+    def setParentJob(self, job, parentId):
```

Comment:
Same here, if we pass in the parent document rather than the ID, it leaves it up to the caller to load the document, which they most likely have already done anyway.